jax/_src/api_util.py

```diff
@@ -132,6 +133,99 @@ def __hash__(self):
   def __eq__(self, other):
     return type(self.val) is type(other.val) and self.val == other.val
 
+_POSITIONAL_ARGUMENTS = (
+  inspect.Parameter.POSITIONAL_ONLY,
+  inspect.Parameter.POSITIONAL_OR_KEYWORD
+)
+
+def validate_argnums(fun: Callable, argnums: Tuple[int, ...], argnums_name: str) -> None:
+  """
+  Validate that the argnums are sensible for a given function.
+
+  Sensibly handles cases of `f(..., *args).`
+  """
+  if min(argnums, default=0) < 0:
+    raise ValueError(f"Jitted function has {argnums_name}={argnums}, "
+                     "but negative argnums are not supported.")
+
+  try:
```

Comment:
Absolutely. Since `_infer_argnums_and_argnames` and the validators don't require the function, I think the best option might be to change the `fun` argument to be the function signature. I'll implement this